### PR TITLE
Group Removal - Part 1

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:* /bcgov/entity#13821

*Description of changes:*
* Part one commit to update Group properties to meet updated API response and schema.
* Removal and Group Refactors to support functionality in Transfers.
* Misc UX Feedback from Transfer Removal Tickets and Draft Tickets
* Fixes MHR and Transfer Submissions

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

File: ppr-ui/src/store/getters/state-getters.ts [hunk 1]
```diff
@@ -725,18 +725,13 @@ export const getMhrTransferHomeOwners = (state: StateIF): MhrRegistrationHomeOwn
   return owners
 }
 
-export const getMhrTransferCurrentHomeOwners = (state: StateIF): MhrRegistrationHomeOwnerGroupIF[] => {
-  const owners = []
+export const getMhrTransferCurrentHomeOwners = (state: StateIF): any => {
```
Comment (dimak1) [inline ppr-ui/src/store/getters/state-getters.ts]: Could we set a _type_ for the return value?

File: ppr-ui/src/store/getters/state-getters.ts [hunk 2]
```diff
@@ -725,18 +725,13 @@ export const getMhrTransferHomeOwners = (state: StateIF): MhrRegistrationHomeOwn
...
+  const ownerGroups = []
+
   state.stateModel.mhrTransfer.currentOwnerGroups.forEach(group => {
```
Comment (dimak1) [inline ppr-ui/src/store/getters/state-getters.ts:731]: If we are not doing anything with the groups then maybe we don't need to iterate and simply return it?

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue [hunk 3]
```diff
@@ -203,35 +205,35 @@ export default defineComponent({
     const parseCurrentOwnerGroups = async (): Promise<void> => {
       const { data } = await fetchMhRegistration(getMhrInformation.value.mhrNumber)
       const currentOwnerGroups = data?.ownerGroups || [] // Safety check. Should always have ownerGroups
+
+      // Store a snapshot of the existing OwnerGroups for baseline of current state
+      await setMhrTransferCurrentHomeOwnerGroups(cloneDeep(data.ownerGroups))
+
       // Create an ID to each individual owner for UI Tracking
       // TODO: Remove after API updates to include the ID for Owners
```
Comment (dimak1) [inline ppr-ui/src/views/mhrInformation/MhrInformation.vue]: This TODO comment is not needed anymore.

**Review by dimak1 — CHANGES_REQUESTED** [review]
Looks good! Just few cleanup requests.

File: ppr-ui/src/store/getters/state-getters.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/store/getters/state-getters.ts]: We can for sure, will update

File: ppr-ui/src/store/getters/state-getters.ts (hunk 2 above)
Comment (cameron-eyds) [reply ppr-ui/src/store/getters/state-getters.ts:731]: Was bumping into an issue here. It wouldn't resolve without an iteration over this. Seemed like a Framework issue, possible a symptom of our in between Vue state. Not sure why but this works.

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Will remove

File: ppr-ui/src/store/getters/state-getters.ts (hunk 1 above)
Comment (cameron-eyds) [reply ppr-ui/src/store/getters/state-getters.ts]: Updated

File: ppr-ui/src/views/mhrInformation/MhrInformation.vue (hunk 3 above)
Comment (cameron-eyds) [reply ppr-ui/src/views/mhrInformation/MhrInformation.vue]: Removed

**Comment by cameron-eyds** [comment]
> Looks good! Just few cleanup requests.

Thanks! Done and done

**Review by dimak1 — APPROVED** [review]

**Review by ChasBoucher — APPROVED** [review]
Great work!!
Looks good.

**Review by doug-lovett — APPROVED** [review]
Looks great, a lot of refactoring.

**Pull request merged**